celest/coordinates/ground_location.py

```diff
@@ -0,0 +1,171 @@
+
+
+from celest.angle_strings import _ISO6709_representation
+from celest.constants import WGS84_MINOR_AXIS_KM, WGS84_MAJOR_AXIS_KM
+from celest.units.quantity import Quantity
+from celest.units.core import Unit
+from celest import units as u
+from math import cos, sin, sqrt
+
+
+class GroundLocation:
+    """GroundLocation(latitude, longitude, height, angular_unit, length_unit)
+
+    Specify Earth bound location.
+
+    The `GroundLocation` class models the Earth with the WGS84 reference
+    ellipsoid to provide accurate representations of ground locations.
+
+    Parameters
+    ----------
+    latitude : float
+        The location's latitude in degrees.
+    longitude : float
+        The location's longitude in degrees.
+    height : float, optional
+        The location's height above the WGS84 ellipsoid.
+    angular_unit : Unit
+        Unit of the `latitude` and `longitude` measures.
+    length_unit : Unit
+        Unit of the `height` measure.
+
+    Attributes
+    ----------
+    latitude : Quantity
+    longitude : Quantity
+    height : Quantity
+    radius : Quantity
+        Earth radius at ground level at the specified position.
+    itrs_x : Quantity
+        Location's X coordinate in the ITRS frame.
+    itrs_y : Quantity
+        Location's Y coordinate in the ITRS frame.
+    itrs_z : Quantity
+        Location's Z coordinate in the ITRS frame.
+    """
+
+    def __init__(self, latitude: float, longitude: float, height: float,
+                 angular_unit: Unit, length_unit: Unit) -> None:
+        """Specify Earth bound location.
+
+        The `GroundLocation` class models the Earth with the WGS84 reference
+        ellipsoid to provide accurate representations of ground locations.
+
+        Parameters
+        ----------
+        latitude : float
+            The location's latitude in degrees.
+        longitude : float
+            The location's longitude in degrees.
+        height : float, optional
+            The location's height above the WGS84 ellipsoid.
+        angular_unit : Unit
+            Unit of the `latitude` and `longitude` measures.
+        length_unit : Unit
+            Unit of the `height` measure.
+        """
+
+        if latitude < -90 or latitude > 90:
```

Comment:
Equality must hold for one of the comparisons. Also consider cascading comparisons. The same is true for the longitude check.